text/BartekJachowicz/opis.tex

```diff
@@ -0,0 +1,129 @@
+\documentclass[12pt]{article}
+\usepackage[utf8]{inputenc}
+\usepackage{polski}
+\usepackage[margin=1in]{geometry}
+\usepackage{amsmath}
+\usepackage{enumitem}
+\usepackage{amsthm}
+\usepackage{amsfonts}
+\usepackage[colorlinks]{hyperref}
+\usepackage{multicol}
+\usepackage{algorithmicx,algpseudocode}
+\usepackage{comment}
+
+\setlength{\parindent}{0pt}
+
+\title{Omówienie pracy\\
+``A Linear Space Algorithm for the LCS Problem``\\
+autorstwa S. Kiran Kumar oraz C. Pandu Rangan
+}
+\author{Bartłomiej Jachowicz}
+\date{}
+
+\theoremstyle{definition}
+\newtheorem{definition}{Definicja}[section]
+
+\theoremstyle{definition}
+\newtheorem{theorem}{Twierdzenie}[section]
+
+\theoremstyle{definition}
+\newtheorem{lemma}{Lemat}[section]
+
+
+\begin{document}
+
+\maketitle
+
+\section{Główna idea algorytmu}
+Algorytm zaprezentowany w pracy opiera się na zastosowaniu techniki \textit{dziel i zwyciężaj}. Pozwala ona znaleźć szukany najdłuższy wspólny podciąg w złożoności $O(n(m - p))$ oraz przy wykorzystaniu liniowego rozmiaru pamięci. W poniższym opisie $A$ oraz $B$ oznaczają słowa o długości odpowiednio $m$ oraz $n$, które są argumentami algorytmu, natomiast $C$ to szukany najdłuższy wspólny podciąg (LCS) o długości $p$. Zapis $T(i:j)$ oznacza podsłowo słowa T z początkiem o indeksie $i$ oraz końcem o indeksie $j$. W opisie pracy zakładam że słowa indeksowane są od cyfry $1$, implementacja algorytmu ze względu praktycznych wykorzystuje indeksowanie słowa od $0$ - przy wywołaniach rekurencyjnych na podsłowach trzeba byłoby przekazywać za każdym razem słowa o jeden znak dłuższe.\\
```

Comment:
Implementation should have an interface which accepts 1-indexed strings. 0-indexed are fine, as long as they are used only internally

Reply:
Done